pywbem_mock/_mainprovider.py

```diff
@@ -1060,11 +1060,33 @@ def ModifyClass(self, namespace, ModifiedClass):
 
             CIMError: CIM_ERR_NOT_SUPPORTED
         """
+        self.validate_namespace(namespace)
 
+        if not isinstance(ModifiedClass, CIMClass):
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("ModifyClass not valid CIMClass. Rcvd type={0}",
+                        type(ModifiedClass)))
+
+        # Validate namespace in class store and get the class CIM repository
+        # for this namespace.
         self.validate_namespace(namespace)
```

Comment:
validate_namespace() is called for the second time.

Reply:
WHOOPS done